src_c/event.c

```diff
@@ -1250,26 +1250,59 @@ dict_from_event(SDL_Event *event)
         case PGE_WINDOWTAKEFOCUS:
         case PGE_WINDOWHITTEST:
         case PGE_WINDOWICCPROFCHANGED:
-        case PGE_WINDOWDISPLAYCHANGED:
-        case SDL_TEXTEDITING:
-        case SDL_TEXTINPUT:
-        case SDL_MOUSEWHEEL:
+        case PGE_WINDOWDISPLAYCHANGED: {
+            window = SDL_GetWindowFromID(event->window.windowID);
+            break;
+        }
+        case SDL_TEXTEDITING: {
+            window = SDL_GetWindowFromID(event->edit.windowID);
+            break;
+        }
+        case SDL_TEXTINPUT: {
+            window = SDL_GetWindowFromID(event->text.windowID);
+            break;
+        }
+        case SDL_DROPTEXT:
+        case SDL_DROPFILE: {
```

Comment:
Would be nice if you also added `SDL_DROPBEGIN` and `SDL_DROPCOMPLETE` here for completeness sake

Reply:
Thanks. I forgot that.

Reply:
![image](https://user-images.githubusercontent.com/31395137/203333657-f9d363ee-6cc0-432a-8672-c5f46b0a6f4f.png)